backtester/engine/grpcserver.go

```diff
@@ -0,0 +1,500 @@
+			if err != nil {
+				return nil, err
+			}
+			mlr, err = decimal.NewFromString(request.Config.CurrencySettings[i].FuturesDetails.Leverage.MaximumLeverageRate)
+			if err != nil {
+				return nil, err
+			}
+			mclr, err = decimal.NewFromString(request.Config.CurrencySettings[i].FuturesDetails.Leverage.MaximumCollateralLeverageRate)
+			if err != nil {
+				return nil, err
+			}
+
+			futuresDetails.Leverage = config.Leverage{
+				CanUseLeverage:                 request.Config.CurrencySettings[i].FuturesDetails.Leverage.CanUseLeverage,
+				MaximumOrdersWithLeverageRatio: mowlr,
+				MaximumOrderLeverageRate:       mlr,
+				MaximumCollateralLeverageRate:  mclr,
+			}
+		}
+
+		configSettings[i] = config.CurrencySettings{
+			ExchangeName:   request.Config.CurrencySettings[i].ExchangeName,
+			Asset:          a,
+			Base:           currency.NewCode(request.Config.CurrencySettings[i].Base),
+			Quote:          currency.NewCode(request.Config.CurrencySettings[i].Quote),
+			SpotDetails:    spotDetails,
+			FuturesDetails: futuresDetails,
+			BuySide: config.MinMax{
+				MinimumSize:  currencySettingBuySideMinimumSize,
+				MaximumSize:  currencySettingBuySideMaximumSize,
+				MaximumTotal: currencySettingBuySideMaximumTotal,
+			},
+			SellSide: config.MinMax{
+				MinimumSize:  currencySettingSellSideMinimumSize,
+				MaximumSize:  currencySettingSellSideMaximumSize,
+				MaximumTotal: currencySettingSellSideMaximumTotal,
+			},
+			MinimumSlippagePercent:        minimumSlippagePercent,
+			MaximumSlippagePercent:        maximumSlippagePercent,
+			MakerFee:                      maker,
+			TakerFee:                      taker,
+			MaximumHoldingsRatio:          maximumHoldingsRatio,
+			SkipCandleVolumeFitting:       request.Config.CurrencySettings[i].SkipCandleVolumeFitting,
+			CanUseExchangeLimits:          request.Config.CurrencySettings[i].UseExchangeOrderLimits,
+			ShowExchangeOrderLimitWarning: request.Config.CurrencySettings[i].UseExchangeOrderLimits,
+			UseExchangePNLCalculation:     request.Config.CurrencySettings[i].UseExchange_PNLCalculation,
+		}
+	}
+
+	var apiData *config.APIData
+	if request.Config.DataSettings.ApiData != nil {
+		apiData = &config.APIData{
+			StartDate:        request.Config.DataSettings.ApiData.StartDate.AsTime(),
+			EndDate:          request.Config.DataSettings.ApiData.EndDate.AsTime(),
+			InclusiveEndDate: request.Config.DataSettings.ApiData.InclusiveEndDate,
+		}
+	}
+	var dbData *config.DatabaseData
+	if request.Config.DataSettings.DatabaseData != nil {
+		cfg := database.Config{
+			Enabled: request.Config.DataSettings.DatabaseData.Config.Enabled,
+			Verbose: request.Config.DataSettings.DatabaseData.Config.Verbose,
+			Driver:  request.Config.DataSettings.DatabaseData.Config.Driver,
+			ConnectionDetails: drivers.ConnectionDetails{
+				Host:     request.Config.DataSettings.DatabaseData.Config.Config.Host,
+				Port:     uint16(request.Config.DataSettings.DatabaseData.Config.Config.Port),
```

Comment:
NOTE for me: This might need more handling because if we define a port that exceeds 16 bits then it will drop a bit and things might go funky.